packages/common/src/wcs/utils/index.ts

```diff
@@ -0,0 +1,288 @@
+import {
+  findTagByName,
+  findTagsByName,
+  findTagsByPath,
+  getAttribute
+} from "xml-utils";
+
+import { findAndParseEnvelope } from "../../gml";
+import { bboxToString, checkExtent, findAndParseCapabilityUrl, findTagText, formatUrl } from "../../common";
+
+import {
+  findAndParseBoundingBox,
+  findAndParseKeywords,
+  findAndParseOperationUrl,
+  findAndParseWGS84BoundingBox,
+} from "../../ows";
+
+import type { BBOX } from "../../types";
+import scaleImage from "../../common/scale-image";
+
+export type Coverage = {
+  id: string;
+  title: string;
+  abstract: string;
+  keywords: string[];
+  bbox: number[];
+};
+
+export function findCoverages(xml: string): string[] {
+  const tagNames = ["CoverageOfferingBrief", "wcs:CoverageSummary"];
+  for (let i = 0; i < tagNames.length; i++) {
+    const tagName = tagNames[i];
+    if (xml.includes(tagName)) {
+      return findTagsByName(xml, tagName).map(tag => tag.outer);
+    }
+  }
+  return [];
+}
+
+export function normalizeCoverageId(id: string): string {
+  return id.replace("__", ":");
+}
+
+export function findCoverageIdentifier(xml: string): string | undefined {
+  return findTagText(xml, "wcs:Identifier");
+}
+
+export function findCoverageId(xml: string): string | undefined {
+  return findTagText(xml,"wcs:CoverageId");
+}
+
+export function findLayerId(
+  xml: string,
+  { normalize = true }: { normalize?: boolean } = { normalize: true }
+): string {
+  // version 2.x
+  const coverageId = findCoverageId(xml);
+  if (coverageId) {
+    return normalize ? normalizeCoverageId(coverageId) : coverageId;
+  }
+
+  // version 1.1.x
+  const identifier = findCoverageIdentifier(xml);
+  if (identifier) return identifier;
+
+  // version 1.0
+  const name = findCoverageName(xml);
+  if (name) return name;
+
+  console.log("===========");
```

Comment:
should this be a `console.debug`?

Reply:
I removed the console.log.  Accidentally left it in from prototyping phase.